src/Symfony/Component/Config/FileLocator.php

```diff
@@ -58,19 +62,28 @@ public function locate($name, $currentPath = null, $first = true)
         $paths = array_unique($paths);
         $filepaths = array();
 
-        foreach ($paths as $path) {
-            if (@file_exists($file = $path.DIRECTORY_SEPARATOR.$name)) {
-                if (true === $first) {
-                    return $file;
-                }
+        $load = function ($file) use (&$filepaths) {
+            if (@file_exists($file)) {
                 $filepaths[] = $file;
             }
+        };
+
+        foreach ($paths as $path) {
+            if (array() !== $files = glob($path.DIRECTORY_SEPARATOR.$name)) {
+                array_walk($files, $load);
+            } else {
+                $load($path.DIRECTORY_SEPARATOR.$name);
```

Comment:
Does this actually do something? Looks like "glob" failed to find any file, so why would "file_exists" (in $load) return something else that "false"? If I'm right, then I'd prefer this array_walk + $load to be replaced by a simple foreach (with no `array() !== ...` check). But I may miss something, that's why I'm asking :)